developer-docs/tips-for-bazel.md

```diff
@@ -92,4 +92,31 @@ $ bazel run //sematic/cli:main -- --help
 You can also do *this* with specific python interpreters:
 ```
 $ bazel run //sematic/cli:main_py39 -- --help
+```
+
+## Troubleshooting
+
+### CPP Toolchain Error
+If you get an error message like:
+
+`"No matching toolchains found for types @bazel_tools//tools/cpp:toolchain_type"`,
+this means bazel can't find your c++ tooling. This can either be due to the fact that
+bazel can't find ANY c++ tooling OR the fact that it can't find c++ tooling suitable
+for building linux docker images.
+
+#### Totally missing c++ tooling
+```
+$ bazel test --toolchain_resolution_debug=@bazel_tools//tools/cpp:toolchain_type //sematic/...
+```
+
+On Macs this is often due to a problem with Xcode. If you are on a Mac:
+- Make sure you have xcode installed
+- Execute `sudo xcodebuild -license` and agree to the license
+- If the above command gives you an error about CommandLineTools and won't show
+the License, try [this](https://stackoverflow.com/a/72115137/2540669)
+
+#### Missing c++ tooling for cross platform docker images
+Just do this:
+```
+$ bazel test --@io_bazel_rules_docker//transitions:enable=false //sematic/...
```

Comment:
I remember I had previously tried this without success, and it still doesn't work, with this patch applied. So please edit the text from `Just do this:` to "try this", and add a disclaimer that if this doesn't work then all is lost and you should use another platform supported by Docker.

```
~/work/sematic$ bazel test --@io_bazel_rules_docker//transitions:enable=false //sematic/...
INFO: Build options --@io_bazel_rules_docker//transitions:enable and --toolchain_resolution_debug have changed, discarding analysis cache.
INFO: Repository sematic-worker-base instantiated at:
  /Users/tudorscurtu/work/sematic/WORKSPACE:151:12: in <toplevel>
  /private/var/tmp/_bazel_tudorscurtu/a689328a338e85f1cbb50b96a7f1fc5e/external/sematic/bazel/pipeline.bzl:148:19: in base_images
Repository rule container_pull defined at:
  /private/var/tmp/_bazel_tudorscurtu/a689328a338e85f1cbb50b96a7f1fc5e/external/io_bazel_rules_docker/container/pull.bzl:294:33: in <toplevel>
ERROR: An error occurred during the fetch of repository 'sematic-worker-base':
   Traceback (most recent call last):
	File "/private/var/tmp/_bazel_tudorscurtu/a689328a338e85f1cbb50b96a7f1fc5e/external/io_bazel_rules_docker/container/pull.bzl", line 240, column 13, in _impl
		fail("Pull command failed: %s (%s)" % (result.stderr, " ".join([str(a) for a in args])))
Error in fail: Pull command failed: 2022/11/11 09:09:02 Running the Image Puller to pull images from a Docker Registry...
2022/11/11 09:09:02 Image pull was unsuccessful: reading image "index.docker.io/sematicai/sematic-worker-base@sha256:d0c0e15f4f20dc60e844523a012c9cc927acbd4c5187b943a4a4a90b0ed70eee": error getting credentials - err: exec: "docker-credential-desktop": executable file not found in $PATH, out: ``
 (/private/var/tmp/_bazel_tudorscurtu/a689328a338e85f1cbb50b96a7f1fc5e/external/go_puller_darwin/file/downloaded -directory /private/var/tmp/_bazel_tudorscurtu/a689328a338e85f1cbb50b96a7f1fc5e/external/sematic-worker-base/image -os linux -os-version  -os-features  -architecture amd64 -variant  -features  -name index.docker.io/sematicai/sematic-worker-base@sha256:d0c0e15f4f20dc60e844523a012c9cc927acbd4c5187b943a4a4a90b0ed70eee)
ERROR: /Users/tudorscurtu/work/sematic/WORKSPACE:151:12: fetching container_pull rule //external:sematic-worker-base: Traceback (most recent call last):
	File "/private/var/tmp/_bazel_tudorscurtu/a689328a338e85f1cbb50b96a7f1fc5e/external/io_bazel_rules_docker/container/pull.bzl", line 240, column 13, in _impl
		fail("Pull command failed: %s (%s)" % (result.stderr, " ".join([str(a) for a in args])))
Error in fail: Pull command failed: 2022/11/11 09:09:02 Running the Image Puller to pull images from a Docker Registry...
2022/11/11 09:09:02 Image pull was unsuccessful: reading image "index.docker.io/sematicai/sematic-worker-base@sha256:d0c0e15f4f20dc60e844523a012c9cc927acbd4c5187b943a4a4a90b0ed70eee": error getting credentials - err: exec: "docker-credential-desktop": executable file not found in $PATH, out: ``
 (/private/var/tmp/_bazel_tudorscurtu/a689328a338e85f1cbb50b96a7f1fc5e/external/go_puller_darwin/file/downloaded -directory /private/var/tmp/_bazel_tudorscurtu/a689328a338e85f1cbb50b96a7f1fc5e/external/sematic-worker-base/image -os linux -os-version  -os-features  -architecture amd64 -variant  -features  -name index.docker.io/sematicai/sematic-worker-base@sha256:d0c0e15f4f20dc60e844523a012c9cc927acbd4c5187b943a4a4a90b0ed70eee)
ERROR: /Users/tudorscurtu/work/sematic/sematic/examples/bazel/BUILD:14:17: //sematic/examples/bazel:bazel_default_image depends on @sematic-worker-base//image:image in repository @sematic-worker-base which failed to fetch. no such package '@sematic-worker-base//image': Pull command failed: 2022/11/11 09:09:02 Running the Image Puller to pull images from a Docker Registry...
2022/11/11 09:09:02 Image pull was unsuccessful: reading image "index.docker.io/sematicai/sematic-worker-base@sha256:d0c0e15f4f20dc60e844523a012c9cc927acbd4c5187b943a4a4a90b0ed70eee": error getting credentials - err: exec: "docker-credential-desktop": executable file not found in $PATH, out: ``
 (/private/var/tmp/_bazel_tudorscurtu/a689328a338e85f1cbb50b96a7f1fc5e/external/go_puller_darwin/file/downloaded -directory /private/var/tmp/_bazel_tudorscurtu/a689328a338e85f1cbb50b96a7f1fc5e/external/sematic-worker-base/image -os linux -os-version  -os-features  -architecture amd64 -variant  -features  -name index.docker.io/sematicai/sematic-worker-base@sha256:d0c0e15f4f20dc60e844523a012c9cc927acbd4c5187b943a4a4a90b0ed70eee)
ERROR: Analysis of target '//sematic/examples/bazel:bazel_default_generate_image_uri' failed; build aborted:
INFO: Elapsed time: 1.393s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (0 packages loaded, 88333 targets configured)
FAILED: Build did NOT complete successfully (0 packages loaded, 88333 targets configured)
    Fetching @local_config_xcode; Building xcode-locator
    Fetching @com_github_google_go_containerregistry; Restarting.
    Fetching @bazel_gazelle_go_repository_config; Restarting.
    Fetching @com_github_pkg_errors; Restarting.
    Fetching @bazel_gazelle_go_repository_tools; fetching
```

Reply:
Yes, curious to know if this solves the problem. Until now I always had to go to the dev box to test cloud submission. Making it work on M1 macs would be amazing.

Reply:
Tudor and I chatted about his issue, and it seems it was due to not having Docker. Added a section about that